src/xapiwrapper.js

```diff
@@ -1369,6 +1367,186 @@ function isDate(date) {
         xhr.send(null);
     }
 
+    var isNode = Boolean(!root.document);
+
+    if (isNode) {
+        // Import encoding-indexes as required by text-encoding module
+        var eIndexes = require('../node_modules/text-encoding/lib/encoding-indexes');
+        global['encoding-indexes'] = eIndexes['encoding-indexes'];
+    }
+
+    // Node shim for browser location
+    var location = isNode ?
```

Comment:
The location object is used in the browser to determine a few things about how the http request should be made. It doesn't exist in node so this is just the minimal necessary replacement to get the code working.